packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview-list.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { Result, XComponentsAdapter } from '@empathyco/x-types';
+import { createLocalVue, mount, Wrapper, WrapperArray } from '@vue/test-utils';
+import Vue from 'vue';
+import {
+  createResultStub,
+  getEmptySearchResponseStub,
+  getResultsStub
+} from '../../../../__stubs__/index';
+import { XComponentsAdapterDummy } from '../../../../__tests__/adapter.dummy';
+import { getDataTestSelector, installNewXPlugin } from '../../../../__tests__/utils';
+import { queriesPreviewXModule } from '../../x-module';
+import QueryPreviewList from '../query-preview-list.vue';
+
+function renderQueryPreviewList({
+  template = '<QueryPreviewList v-bind="$attrs"/>',
+  queries = ['milk'],
+  results = { milk: getResultsStub(1) }
+}: RenderQueryPreviewListOptions): RenderQueryPreviewListApi {
+  const localVue = createLocalVue();
+  const adapter: XComponentsAdapter = {
+    ...XComponentsAdapterDummy,
+    search({ query }) {
+      const fakeResults = results[query];
+      return Promise.resolve({
+        ...getEmptySearchResponseStub(),
+        results: fakeResults,
+        totalResults: fakeResults.length
+      });
+    }
+  };
+  installNewXPlugin({ initialXModules: [queriesPreviewXModule], adapter }, localVue);
+  const wrapper = mount(
+    {
+      template,
+      components: {
+        QueryPreviewList
+      }
+    },
+    {
+      localVue,
+      propsData: {
+        queries
+      }
+    }
+  );
+  return {
+    wrapper: wrapper.findComponent(QueryPreviewList),
+    getQueryPreviewItemWrappers() {
+      return wrapper.findAll(getDataTestSelector('query-preview-item'));
+    },
+    reRender() {
+      // A timeout with no time should resolve after all reactivity + promises involved
+      return new Promise(resolve => setTimeout(resolve));
+    }
+  };
+}
+
+describe('testing QueryPreviewList', () => {
+  it('renders a list of queries one by one', async () => {
+    const { getQueryPreviewItemWrappers, reRender } = renderQueryPreviewList({
+      template: `
```

Comment:
we should use this template as the default one, we are using the same template in both tests.